main.go

```diff
@@ -224,49 +228,132 @@ func main() {
 	}
 
+	validFilterOptions := []string{"uid"}
+
+	filter := tracee.Filter{
+		UIDFilter: baseUIDFilter(),
+	}
+
 	for _, f := range filters {
-		s := strings.Split(f, "=")
-		if len(s) != 2 {
-			return tracee.Filter{}, fmt.Errorf(filterHelp)
+
+		// Parse the filter option (i.e. uid, ...)
+		isValid := false
+		filterSlice := []string{}
+		for i := range validFilterOptions {
+			if strings.HasPrefix(f, validFilterOptions[i]) {
+				filterSlice = strings.SplitAfter(f, validFilterOptions[i])
+				isValid = true
+			}
 		}
-		if !validFilterOption(s[0]) {
-			return tracee.Filter{}, fmt.Errorf("invalid filter: %s\n%s", s[0], filterHelp)
+		if !isValid {
+			return tracee.Filter{}, fmt.Errorf("invalid filter option specified, use '--filter help' for more info")
 		}
 
-		if s[0] == "uid" {
-			values := strings.Split(s[1], ",")
-			for _, v := range values {
-				uid, err := strconv.ParseUint(v, 10, 32)
-				if err != nil {
-					return tracee.Filter{}, fmt.Errorf("specified invalid uid: %s", v)
-				}
-				uids = append(uids, uint32(uid))
+		if filterSlice[0] == "uid" {
+			err := parseUIDFilter(filterSlice[1], filter.UIDFilter)
+			if err != nil {
+				return tracee.Filter{}, err
 			}
 		}
 	}
-	return tracee.Filter{
-		UIDs: uids,
-	}, nil
+
+	return filter, nil
+}
+
+// Operator represents a comparison operator such as '=', '!=', '>', or '<'
+type Operator uint8
+
+const (
+	opInvalid Operator = iota
+	opEqual
+	opNotEqual
+	opGreater
+	opLess
+)
+
+func parseUIDFilter(operatorAndValues string, uidFilter *tracee.UIDFilter) error {
+	validFilterOperators := []string{"=", "!=", ">", "<"}
+	isValid := false
+	valueSlice := []string{}
+	for i := range validFilterOperators {
+		if strings.HasPrefix(operatorAndValues, validFilterOperators[i]) {
+			valueSlice = strings.SplitAfter(operatorAndValues, validFilterOperators[i])
+			isValid = true
+		}
+	}
+	if !isValid {
+		return fmt.Errorf("invalid filter operator specified, use '--filter help' for more info")
+	}
+
+	//valueSlice is ["<operator>", "<uid>", ...]
+	operator := parseOperator(valueSlice[0])
+	if operator == opInvalid {
+		return fmt.Errorf("invalid filter operator: %s", valueSlice[0])
+	}
+
+	values := strings.Split(valueSlice[1], ",")
+
+	for i := range values {
+		v, err := strconv.ParseUint(values[i], 10, 32)
+		if err != nil {
+			return fmt.Errorf("invalid UID value: %s", values[i])
+		}
+		if operator == opEqual {
+			uid := uint32(v)
+			uidFilter.Equal = append(uidFilter.Equal, uid)
+		}
+		if operator == opNotEqual {
+			uid := uint32(v)
+			uidFilter.NotEqual = append(uidFilter.NotEqual, uid)
+		}
+		uid := int64(v)
+		if operator == opGreater && uid < uidFilter.Greater {
+			uidFilter.Greater = uid
+		}
+		if operator == opLess && uid > uidFilter.Less {
+			uidFilter.Less = uid
+		}
+	}
+
+	return nil
+}
+
+func baseUIDFilter() *tracee.UIDFilter {
+	return &tracee.UIDFilter{
+		Equal:    []uint32{},
+		NotEqual: []uint32{},
+		Greater:  int64(math.MaxUint32 + 1),
```

Comment:
Don't we need Greater to be equal to "-1-" and Less be equal to (math.MaxUint32 + 1) so defaults will always evaluate to true?

Reply:
This relates to my comment above. I have it set to to closed by default so when reading in user-supplied filter options I can compare based on it being lower. When loading into BPF it compares to this default value and won't load into the BPF map at all if it hasn't changed. I can swap it around if you'd like.

Reply:
If that works, and you think it's more intuitive, I don't mind keeping it this way